variables.tf

```diff
@@ -0,0 +1,274 @@
+# VPC
+#################
+variable "create_vpc" {
+  description = "Whether to create vpc. If false, you can specify an existing vpc by setting 'vpc_id'."
+  type        = bool
+  default     = true
+}
+
+variable "existing_vpc_id" {
+  description = "The vpc id used to launch several vswitches."
+  type        = string
+  default     = ""
+}
+
+variable "use_existing_vpc" {
+  description = "The vpc id used to launch several vswitches. If set, the 'create_vpc' will be ignored."
+  type        = bool
+  default     = false
+}
+
+variable "vpc_name" {
+  description = "The vpc name used to launch a new vpc."
+  type        = string
+  default     = "TF-VPC"
+}
+
+variable "vpc_cidr" {
+  description = "The cidr block used to launch a new vpc."
+  type        = string
+  default     = "172.16.0.0/12"
+}
+
+variable "vpc_tags" {
+  description = "The tags used to launch a new vpc. Before 1.5.0, it used to retrieve existing VPC."
+  type        = map(string)
+  default     = {}
+}
+
+variable "tags" {
+  description = "The common tags will apply to all of resources."
+  type        = map(string)
+  default     = {}
+}
+
+#################
+# Vswitch
+#################
+variable "vswitch_cidrs" {
+  description = "List of cidr blocks used to launch several new vswitches. If not set, there is no new vswitches will be created."
+  type        = list(string)
+  default     = []
+}
+
+variable "availability_zones" {
+  description = "List available zones to launch several VSwitches."
+  type        = list(string)
+  default     = []
+}
+
+variable "vswitch_name" {
+  description = "The vswitch name prefix used to launch several new vswitches."
+  default     = "TF-VSwitch"
+}
+
+variable "vswitch_tags" {
+  description = "The tags used to launch serveral vswitches."
+  type        = map(string)
+  default     = {}
+}
+
+########################
+#variables for nat gateway
+########################
+variable "create_nat" {
+  description = "Whether to create nat gateway. If false, you can specify an existing nat gateway by setting 'nat_gateway_id'."
+  type        = bool
+  default     = true
+}
+
+variable "nat_name" {
+  description = "Name of a new nat gateway."
+  type        = string
+  default     = "terraform-alicloud-nat-gateway"
+}
+
+variable "nat_specification" {
+  description = "The specification of nat gateway."
+  type        = string
+  default     = "Small"
+}
+
+variable "nat_instance_charge_type" {
+  description = "The charge type of the nat gateway. Choices are 'PostPaid' and 'PrePaid'."
+  type        = string
+  default     = "PostPaid"
+}
+
+variable "nat_period" {
+  description = "The charge duration of the PrePaid nat gateway, in month."
+  type        = number
+  default     = 1
+}
+
+########################
+# New EIP parameters
+########################
+variable "create_eip" {
+  description = "Whether to create new EIP and bind it to this Nat gateway. If true, the 'number_of_eip' should not be empty."
+  type        = bool
```

Comment:
If true, the 'number_of_dnat_eip' or  'number_of_snat_eip'  should not be empty.